napari_aicssegmentation/widgets/workflow_thumbnails.py

```diff
@@ -0,0 +1,143 @@
+
+        self.step_3_instructions.setObjectName("step3InstructionsDisabled")
+        self.layout().addWidget(step_3)
+
+        # Row of text labeling the columns of workflow images
+        self.column_labels = QWidget()
+        column_layout = QHBoxLayout()
+        column_layout.setContentsMargins(11, 11, 11, 0)
+        self.column_labels.setLayout(column_layout)
+
+        image_input_label = QLabel("Image input")
+        image_input_label.setAlignment(QtCore.Qt.AlignCenter)
+        segmentation_output_label = QLabel("Segmentation output")
+        segmentation_output_label.setAlignment(QtCore.Qt.AlignCenter)
+        self.column_labels.layout().addWidget(image_input_label)
+        self.column_labels.layout().addWidget(segmentation_output_label)
+
+        self.column_labels.setFixedWidth(PAGE_CONTENT_WIDTH)
+        self.column_labels.setObjectName("columnLabelsDisabled")
+        self.layout().addWidget(self.column_labels, alignment=QtCore.Qt.AlignCenter)
+
+    def _add_buttons(self, workflows: List[WorkflowDefinition]):
+        """
+        Add all buttons given a List of WorkflowDefinitions
+        """
+
+        for workflow in workflows:
+            # Some images are RGBA and others are Grayscale
+            # TODO?: convert all images to RBGA
+            pre: np.ndarray = workflow.thumbnail_pre
+            post: np.ndarray = workflow.thumbnail_post
+            # If RGBA convert to grayscale
+            if len(workflow.thumbnail_pre.shape) > 2:
+                # cv2 expects color channel dim to be last index
+                pre = cv2.cvtColor(np.moveaxis(workflow.thumbnail_pre, 0, -1), cv2.COLOR_RGBA2GRAY)
+            if len(workflow.thumbnail_post.shape) > 2:
+                # cv2 expects color channel dim to be last index
+                post = cv2.cvtColor(np.moveaxis(workflow.thumbnail_post, 0, -1), cv2.COLOR_RGBA2GRAY)
+            # Stitch Image
+            image_stitched: np.ndarray = np.hstack([pre, post])
+            button: QPushButton = QPushButton("")
+            # Get np image into QPixmap
+            image: QPixmap = QPixmap(
+                QImage(image_stitched.data, image_stitched.shape[1], image_stitched.shape[0], QImage.Format_Indexed8)
+            )
+            button.setIcon(QIcon(image))
+            button.setIconSize(QtCore.QSize(PAGE_CONTENT_WIDTH - 40, 200))
+            button.setFixedSize(PAGE_CONTENT_WIDTH, 200)
+
+            button.setEnabled(False)
+
+            self.layout().addWidget(button)
+
+    def setEnabled(self, enabled: bool) -> None:
+        super().setEnabled(enabled)
+        if enabled:
+            self.column_labels.setObjectName("columnLabels")
+            self.step_3_instructions.setObjectName("step3Instructions")
+            self.setStyleSheet(Style.get_stylesheet("main.qss"))
+
+            self._enable_buttons()
+        else:
+            self.column_labels.setObjectName("columnLabelsDisabled")
```

Comment:
Not a big deal here, but for these types of "style toggle" features - it's usually easier to just wrap all the widgets in a Frame or any kind of parent widget - and then only worry about toggling the CSS class on the parent widget. Then in your CSS you just use selectors, ie.
```
#workflowsDisabled QLabel{...}
#workflowsDisabled #columnLabels{...}
```

Reply:
Yeah I think the tricky bit about this specific case is that there is also the QLabel for the selection step number "3." involved in the same Form row as the instructions, which we don't want to disable.